gnuradio-runtime/lib/hier_block2_detail.cc

```diff
@@ -496,9 +496,27 @@ endpoint_vector_t hier_block2_detail::resolve_port(int port, bool is_input)
     return result;
 }
 
+void hier_block2_detail::recursive_disconnect_all(const std::string& caller)
```

Comment:
why do we have to identify the caller it by its string identifier? Why can't we simply use the pointer?